spec/need_kconfig_format_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+require 'yaml'
+
+def check_config_name(name)
+  # Check if config name format is valid:
+  # 1. not start with CONFIG_
+  !name.to_s.start_with?('CONFIG_')
+end
+
+def check_config_value(value)
+  # Check if value format is valid:
+  # 1. only contain n, m, y or nothing
+  # 2. only contain i386, x86_64, or nothing
+  allowed_chars = /^(n|m|y|i386|x86_64|)$/
+
+  # Convert value to string
+  value = value.to_s
+
+  # If the value is an array, check each element seperately
+  if value.is_a?(Array)
+    value.all? { |v| v.match(allowed_chars) || v.empty? }
+  else
+    value.match(allowed_chars) || value.empty?
+  end
```

Comment:
probably the code can be combined as
```
Array(value).all? { |v| v.match(allowed_chars) || v.empty? }
```

But would value.is_a?(Array) become true, since early code has converted the value to String.